opentelemetry-sdk/src/attributes/set.rs

```diff
@@ -104,37 +105,46 @@ impl Eq for HashKeyValue {}
 ///
 /// This must implement [Hash], [PartialEq], and [Eq] so it may be used as
 /// HashMap keys and other de-duplication methods.
-#[derive(Clone, Default, Debug, Hash, PartialEq, Eq)]
-pub struct AttributeSet(BTreeSet<HashKeyValue>);
+#[derive(Clone, Default, Debug, PartialEq, Eq)]
+pub struct AttributeSet(Vec<HashKeyValue>, u64);
 
 impl From<&[KeyValue]> for AttributeSet {
     fn from(values: &[KeyValue]) -> Self {
-        let mut seen = HashSet::with_capacity(values.len());
-        AttributeSet(
-            values
-                .iter()
-                .rev()
-                .filter_map(|kv| {
-                    if seen.contains(&&kv.key) {
-                        None
-                    } else {
-                        seen.insert(&kv.key);
-                        Some(HashKeyValue(kv.clone()))
-                    }
-                })
-                .collect(),
-        )
+        let mut seen_keys = HashSet::with_capacity(values.len());
+        let mut vec = values
+            .iter()
+            .rev()
+            .filter_map(|k| {
+                if seen_keys.insert(k.key.clone()) {
+                    Some(HashKeyValue(k.clone()))
+                } else {
+                    None
+                }
+            })
+            .collect::<Vec<_>>();
+        vec.sort();
```

Comment:
nit: Think we can have a small optimization here as we don't need stable sorting for items without duplicate keys?

```suggestion
        vec.sort_unstable();
```

Reply:
Good call. changed!